yt/frontends/enzo_e/fields.py

```diff
@@ -143,17 +143,34 @@ def setup_particle_mass_field(self, ptype):
                 constants = (constants,)
             names = [c[0] for c in constants]
 
+        group_list = nested_dict_get(
+            self.ds.parameters, ("Particle", ptype, "group_list"), default=()
+        )
+
         if "mass" in names:
-            val = constants[names.index("mass")][2]
-            val = self.ds.quan(val, self.ds.mass_unit)
-            if self.ds.cosmological_simulation:
+            # Prior to the addition of the is_gravitating group list,
+            # particle masses were stored as densities if the simulation
+            # was cosmological.
+            pfield = "mass"
+            is_density = (
+                "is_gravitating" not in group_list and self.ds.cosmological_simulation
+            )
```

Comment:
The "cosmological simulation" condition should be removed here.

Reply:
Now that I've thought about this, I'm not sure that checking for the `"is_gravitating"` particle group is the most robust solution. The reason is that it requires the user to define the `"is_gravitating"` group in the input parameter file whenever they have particles with mass, even if they are not running with gravity, which is a bit counter-intuitive. I think a better solution would be to have a hard-coded parameter called something like "mass_is_mass", which would be guaranteed to exist if the data was produced with a version of Enzo-E built after the change in definition of "mass". The question I have is: how exactly does yt read in Enzo-E parameters?

Reply:
Parameters are mainly read in from the .libconfig file. We could, in principle, also look for HDF5 attributes in the data files.